### PR TITLE
feat(search): use tf serve API from Python

### PR DESCRIPTION
**Pull request by rmelisson**
#2750

### CODE REVIEW

File: .k8s/api.values.yml [hunk 1]
```diff
@@ -101,7 +101,7 @@ deployment:
         - |
           retry=120; # 5s * (12 * 10) = 10min
           while
-            ! curl -sS --fail "${NLP_URL}" &&
+            ! true &&
```
Comment (douglasduteil) [inline .k8s/api.values.yml]: Is this init container still relevant ?

File: README-dev.md [hunk 2]
```diff
@@ -99,17 +98,15 @@ Build a local NLP image
...
-docker run --rm --name cdtn-nlp -p 5000:5000 cdtn_nlp:local
-```
 
-Create a dump with semantic vectors (you will need a NLP service running)
+Create a dump with semantic vectors (you will need a NLP service available)
 (if NLP_URL env is not provide it will create a dump without semantic vectors)
 
 ```
-NLP_URL=http://localhost:5000 yarn workspace @cdt/data dump-dev
+NLP_URL=https://serving-ml.prod2.fabrique.social.gouv.fr/ yarn workspace @cdt/data dump-dev
```
Comment (douglasduteil) [inline README-dev.md]:
```suggestion
NLP_URL=https://serving-ml.fabrique.social.gouv.fr/ yarn workspace @cdt/data dump-dev
```

File: packages/code-du-travail-api/package.json [hunk 3]
```diff
@@ -62,4 +62,4 @@
       "lib/**/*.js"
     ]
   }
-}
+}
```
Comment (douglasduteil) [inline packages/code-du-travail-api/package.json]:
```suggestion
}

```

File: packages/code-du-travail-data/README.md [hunk 4]
```diff
@@ -16,11 +16,11 @@ Les données utilisées par le site sont issues de packages de données (@social
 $ yarn workspace @cdt/data dump-dev
 ```
 
-Pour chaque document, il possible aussi de rajouter une représentation vectorielle qui servira à la recherche semantique. Pour cela, il faut que la variable d'environnement NLP_URL soit fournie et pointe vers une instance du service NLP. Reportez-vous au [README.md du projet nlp](../code-du-travail-nlp/README.md) pour voir comment démarrer une instance du service.
+Pour chaque document, il possible aussi de rajouter une représentation vectorielle qui servira à la recherche semantique. Pour cela, il faut que la variable d'environnement NLP_URL soit fournie et pointe vers une instance du service NLP. Reportez-vous au [README.md du projet serving-ml](https://github.com/SocialGouv/serving-ml) pour voir comment démarrer une instance locale du service. Une instance est également accessible à cette URL : https://serving-ml.prod2.fabrique.social.gouv.fr/
```
Comment (douglasduteil) [inline packages/code-du-travail-data/README.md]:
```suggestion
Pour chaque document, il possible aussi de rajouter une représentation vectorielle qui servira à la recherche semantique. Pour cela, il faut que la variable d'environnement NLP_URL soit fournie et pointe vers une instance du service NLP. Reportez-vous au [README.md du projet serving-ml](https://github.com/SocialGouv/serving-ml) pour voir comment démarrer une instance locale du service. Une instance est également accessible à cette URL : https://serving-ml.fabrique.social.gouv.fr/ 
```

File: packages/code-du-travail-data/dataset/stop_words/semantic_stop_words.json [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+    "lui",
+    "ne",
+    "avez",
+    "seront",
+    "tous",
+    "ait",
+    "ceux",
+    "ta",
+    "seraient"
+]
```
Comment (douglasduteil) [inline packages/code-du-travail-data/dataset/stop_words/semantic_stop_words.json]:
```suggestion
]

```

File: packages/code-du-travail-data/indexing/vectorizer.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
+// import fetch from "node-fetch";
+// import semantic_stopwords from "../dataset/stop_words";
+const fetch = require("node-fetch");
+const semantic_stopwords = require("../dataset/stop_words");
```
Comment (douglasduteil) [inline packages/code-du-travail-data/indexing/vectorizer.js]: Did you switch from ES modules to CommonJS for a reason ?

**Review by douglasduteil — CHANGES_REQUESTED** [review]
- might want to remove the `wait-for-nlp` init container on the `.k8s/api.values.yml#L89-110`

File: packages/code-du-travail-data/indexing/vectorizer.js (hunk 6 above)
Comment (rmelisson) [reply packages/code-du-travail-data/indexing/vectorizer.js]: code-du-travail-api is using CommonJS, and throwing an exception when requiring code-du-travail-data ES module, will see if I can find a work around

File: .k8s/api.values.yml (hunk 1 above)
Comment (rmelisson) [reply .k8s/api.values.yml]: I think so, I've just updated the command to the check if the nlp service is alive.

File: .vscode/settings.json [hunk 7]
```diff
@@ -0,0 +1,3 @@
+{
+    "git.ignoreLimitWarning": true
+}
```
Comment (douglasduteil) [inline .vscode/settings.json]: Is this file necessary ?

File: .yarnclean [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+.*ignore
+.eslintrc
+.jshintrc
+.flowconfig
+.documentup.json
+.yarn-metadata.json
+.travis.yml
+
+# misc
+*.md
```
Comment (douglasduteil) [inline .yarnclean:45]: Is this file necessary ?

File: packages/code-du-travail-data/indexing/vectorizer.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+  const noStopWords = split.filter((t) => !stopWords.has(t.toLowerCase()));
+
+  return noStopWords.join(" ");
+}
+
+async function callTFServe(body) {
+  const response = await fetch(tfServeURL, { method: "POST", body });
+  if (response.ok) {
+    const json = await response.json();
+    return json["outputs"];
```
Comment (douglasduteil) [inline packages/code-du-travail-data/indexing/vectorizer.js:34]:
```suggestion
    return json.outputs;
```

**Review by douglasduteil — CHANGES_REQUESTED** [review]
Is `packages/code-du-travail-data/error.log` necessary ?

File: .yarnclean (hunk 8 above)
Comment (rmelisson) [reply .yarnclean:45]: my bad

**Review by douglasduteil — APPROVED** [review]

**Comment by douglasduteil** [comment]
Good work @rmelisson :)
